playwright-e2e/pages/respondents-details.ts

```diff
@@ -0,0 +1,68 @@
+import { type Page, type Locator, expect } from "@playwright/test";
+import test from "node:test";
+
+export class RespondentsDetails {
+    readonly page: Page;
+    readonly RespondentsDetailsLink: Locator;
+    readonly RespondentsDetailsHeading: Locator;
+    readonly LegalCounselTestCase: Locator;
+    readonly AddNew: Locator;
+    readonly FirstName: Locator;
+    readonly LastName : Locator;
+    readonly DateOfBirth : Locator;
+    readonly DayMonthYear : Locator;
+    readonly Gender : Locator;
+    readonly CurrentAddressKnown: Locator
+    readonly RelationshipToChildLocator : Locator;
+    readonly DoYouNeedContactDetailsHiddenFromParties : Locator;
+    readonly AbilityToTakePartInProceedings: Locator;
+    readonly DoTheyHaveLegalRepresentation: Locator;
+    readonly RespondentsDetailsNeeded: Locator;
+  static RespondentsDetailsHeading: any;
+    
+  public constructor(page: Page) {
+        this.page = page;
+        this.RespondentsDetailsHeading = page.getByRole("heading", {name: "AllocationProposal needed",});
+        this.RespondentsDetailsLink = page.locator(".govuk-template__body.js-enabled");
+        this.RespondentDetailsNeeded(); 
+}
+async RespondentDetailsNeeded() {
+  await this.page.getByRole('heading', { name: 'Add information about the' }).isVisible();
+  await this.page.getByRole('link', { name: 'Respondents\' details' }).click();
+  await this.page.getByLabel('*First name (Optional)').fill('Asa');
+  await this.page.getByLabel('*Last name (Optional)').click();
+  await this.page.getByLabel('*Last name (Optional)').fill('Thierry');
+  await this.page.getByText('Date of birth (Optional)').click();
+  await this.page.getByLabel('Day').isVisible();
+  await this.page.pause();
+  await this.page.getByLabel('Day').fill('20');
+  await this.page.getByLabel('Month').click();   
```

Comment:
Day, month, year, gender etc. can be defined just once and reused

Reply:
True